pkg/gauges/contracts/AuthorizerAdaptor.sol

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/IAuthentication.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/Address.sol";
+
+import "@balancer-labs/v2-vault/contracts/interfaces/IAuthorizer.sol";
+import "@balancer-labs/v2-vault/contracts/interfaces/IVault.sol";
+
+/**
+ * @title Authorizer Adaptor
+ * @notice This contract is intended to act as an adaptor between systems which expect a single admin address
+ * and the Balancer Authorizer such that the Authorizer may grant/revoke admin powers to unlimited addresses. 
+ *
+ * The permissions the Authorizer can grant are granular such they may be global or specific to a particular contract
+ * @dev When calculating the actionId to call a function on a target contract, it must be calculated as if it were 
+ * to be called on this adaptor. This can be done by passing the function selector to the `getActionId` function.
+ */
+contract AuthorizerAdaptor is IAuthentication {
+    using Address for address;
+
+    bytes32 private immutable _actionIdDisambiguator;
+    IVault private immutable _vault;
+
+    constructor(IVault vault) {
+        // AuthorizerAdaptor is a singleton, so it simply uses its own address to disambiguate action identifiers
+        _actionIdDisambiguator = bytes32(uint256(address(this)));
+        _vault = vault;
+    }
+
+    /**
+     * @notice Returns the Balancer Vault
+     */
+    function getVault() public view returns (IVault) {
+        return _vault;
+    }
+
+    /**
+     * @notice Returns the Authorizer
+     */
+    function getAuthorizer() external view returns (IAuthorizer) {
+        return _getAuthorizer();
+    }
+
+    function _getAuthorizer() internal view returns (IAuthorizer) {
+        return getVault().getAuthorizer();
+    }
```

Comment:
```suggestion
    function getAuthorizer() public view returns (IAuthorizer) {
        return getVault().getAuthorizer();
    }
```

Reply:
We do have that internal function in some contracts of the Authorization tree, but I'd keep this one simple.